src/components/map/RouteLayer.tsx

```diff
@@ -34,7 +47,47 @@ export default class RouteLayer extends Component<RouteLayerProps> {
             !== this.props.routePaths.map(rPath => rPath.internalId).join(':');
         if (routePathsChanged) {
             this.calculateBounds();
+            this.setState({
+                selectedPolylines: [],
+            });
+        }
+    }
+
+    private selectPolyLine(routePath: IRoutePath) {
+        const selectedPolylines = this.state.selectedPolylines;
+        const indexInArray: number = selectedPolylines.indexOf(routePath.internalId);
+
+        if (indexInArray > -1) {
+            selectedPolylines.splice(indexInArray, 1);
+        } else {
+            selectedPolylines.push(routePath.internalId);
+        }
+
+        this.setState({
+            selectedPolylines,
+        });
+    }
+
+    private isSelected(internalId: string) {
+        if (this.state.selectedPolylines.includes(internalId) ||
+            this.state.hoveredPolylines.includes(internalId)) {
+            return true;
         }
+        return false;
+    }
+
+    private mouseHoverIn(internalId: string) {
+        const hoveredPolylines = this.state.hoveredPolylines;
+        hoveredPolylines.push(internalId);
+        this.setState({
+            hoveredPolylines,
+        });
```

Comment:
This could be slightly shorter with:

```
private mouseHoverIn(internalId: string) {
    this.setState({
        hoveredPolylines: this.state.hoveredPolylines.concat(internalId),
    });
}
```

Reply:
But this is just a tip, the current code works equally well, and might well be more clearer. So, do as you want.